src/store/modules/wallet.js

```diff
@@ -569,56 +569,53 @@ const actions = {
     // 4. group batch send events
     const batchSendEventsMap = new Map();
 
-    const isPotentialBatchSendEvent = ({ type, sender, txHash }) =>
+    const isEligibleForBatchSendMap = ({ type, sender, txHash }) =>
       type === 'send' &&
       followees.includes(sender) &&
       !batchSendEventsMap.has(txHash);
 
-    const isBatchSendEvent = ({ type, sender, txHash }) =>
+    const isAlreadyInBatchSendMap = ({ type, sender, txHash }) =>
       type === 'send' &&
       followees.includes(sender) &&
       batchSendEventsMap.has(txHash);
 
     filteredEvents.forEach(event => {
       const { type, sender, tx_hash: txHash, receiver } = event;
-      if (isPotentialBatchSendEvent({ type, sender, txHash })) {
+      const storedBatchSendEvent = batchSendEventsMap.get(txHash);
+
+      if (isEligibleForBatchSendMap({ type, sender, txHash })) {
         batchSendEventsMap.set(txHash, { ...event, batchSendList: [receiver] });
-      } else if (isBatchSendEvent({ type, sender, txHash })) {
-        const storedBatchSendEvent = batchSendEventsMap.get(txHash);
+      } else if (isAlreadyInBatchSendMap({ type, sender, txHash })) {
         if (!storedBatchSendEvent.batchSendList.includes(receiver)) {
           storedBatchSendEvent.batchSendList.push(receiver);
         }
       }
-      if (filteredEvents?.length) {
-        filteredEvents.forEach(event => {
-          const existingEvents = state.followeeEventsMap[event.tx_hash];
-          commit(WALLET_SET_FOLLOWEE_EVENTS, {
-            txHash: event.tx_hash,
-            event: { ...existingEvents, ...event },
-          });
+
+      // 5.commit batch send events
+      const isBatchSendEvent =
+        storedBatchSendEvent && storedBatchSendEvent.batchSendList?.length > 1;
+      if (isBatchSendEvent) {
+        commit(WALLET_SET_FOLLOWEE_EVENTS, {
+          txHash,
+          event: {
+            ...event,
+            batchSendList: storedBatchSendEvent.batchSendList,
+          },
+        });
```

Comment:
Since we are inside the event forEach loop, the batchsend event would be incomplete and commited more than once
Should make the commit part outside of the loop, like in another loop i suppose